core/ir/opnd_api.h

```diff
@@ -1820,6 +1820,12 @@ struct _opnd_t {
         /* Used for ARM: REG_kind, BASE_DISP_kind, and IMMED_INTEGER_kind */
         ushort /*dr_opnd_flags_t*/ flags;
     } aux;
+#    ifdef RISCV64
+    /* Used by disassemble, indicates whether an immediate value should be formatted
+     * in decimal or hexadecimal, valid iff opnd is an IMMED_INTEGER_kind.
+     */
+    bool decimal;
```

Comment:
In what operands is this set to true vs false. Is there any semantic property we can attach to this? If yes, then this may be better off as a new dr_opnd_flags_t bit.

Reply:
I think it's only a format thing, i.e. make the disassembled string looks better.

Reply:
I mean -- for what operands did you decide to use the hex format vs the decimal format.

Reply:
Basically, if an immediate is used as an arithmetic operand (like addi, slli), or as an offset (like load, store), it is better displayed as a decimal, as objdump does. It would be better to add a new dr_opnd_flags_t bit for this indeed.

Reply:
But I didn't quite think of the name of this flag.

Reply:
I think it'd be reasonable to just add something like DR_OPND_IMM_PRINT_DECIMAL. @derekbruening any thoughts?

Reply:
In general disassembly format has not historically had first-class support in DR's decoder.  For example, on x86 to support Intel or AT&T assembler format conventions we have code on top of the actual decode tables that kind of hackily and imperfectly applies those format conventions.  We've thought about giving disassembly in standard assembler styles first-class support: usually our idea has been to add a printf string template to each decode table entry (xref #1683 on this for AArch32 and #1307 on x86), but it's never been a high priority.  That said, if it can be done without any drawbacks to DR's normal operation (i.e., without increasing IR data type sizes or adding IR overhead), then I'm all for it.

A dr_opnd_flags_t bit seems like a reasonable solution; an alternative solution would be a printf string template per opcode encoding.  Are there other printing decisions that would need other IR changes, while a printf template would solve them all at once?  Or is this decimal-vs-hex the only foreseen issue?

I wonder if any other arches would want to use this dr_opnd_flags_t bit -- seems like for other arches it might be part of a runtime option for numeric print preference.

Reply:
In addition to the display format of immediate data, known CSR registers may be better displayed by the name of the register. And fallback to hex decimals for unknown CSRs. Currently, all CSRs are displayed as hex.

Reply:
Added `DR_OPND_IMM_PRINT_DECIMAL` enum in https://github.com/DynamoRIO/dynamorio/pull/6095/commits/2f2d61574fbd3569e463c11555389b17009d1642.